controllers/mediaController.js

```diff
@@ -0,0 +1,60 @@
+// import Media schema from models
+const { User } = require('../models/mainModels.js');
+// destructured media property of User schema
+const { media } = User;
+
+
+// object contains all mediaController middleware
+const mediaController = {};
+
+
+// middleware to get entire media profile of user
+mediaController.getMedia = (req, res, next) => {
+  // find all media catalog
+  User.findOne({_id:req.query.userId}).exec()
+    .then(response => {
+      // console.log to see if fired
+      console.log("mediaController getMedia fired")
+      // store in local memory
+    res.locals.media = response;
+    return next();
+  })
+  .catch(err => {
+    next('error in getMedia middleware');
+  })
+};
+
+// add a media type to user profile
+mediaController.addMedia = (req, res, next) => {
+//media.create, create conditions for data to allow into array
+const { title, type, currentStatus } = req.body;
+  
+  // update media property on user
+  User.update({ title, type, currentStatus })
```

Comment:
Going back to my notes, I think this should specifically update the media property of the user document with req.body.

Don't worry about destructuring the req.body. The front end will have to send the title, type, currentStatus (AND maybe other fields too) so you guys can just send the req.body directly into the media property of the document. The media schema will enforce whether the front end included the title/type/currentStatus - mongoose will return an error, I think, into the catch block if the request body doesn't have the right fields.

Reply:
A possible approach (syntax is not exactly correct - this is the general idea)

User.findOne({_id: req.params.userId}).
  then -> userMedia = user.media
userMedia.push(req.body)

User.update({_id: req.pramsId}, {media: userMedia})

user.media.push(req.body)

Reply:
Added $push and $pull methods for update but not sure what the syntax for the mediaId we push into the array is